src/wasm-type.h

```diff
@@ -470,6 +476,14 @@ struct Signature {
   std::string toString() const;
 };
 
+struct Continuation {
+  HeapType ht;
```

Comment:
Naive question from someone unfamiliar with the Type Continuations spec: If a continuation is just a heap type, why do we need a Continuation type?

Reply:
From a spec perspective, continuation types are built from existing function types, but they are different from them: Given an already defined function type `$f`, we have that `(cont $f)` is the type of continuations whose signature is like that of `$foo`, but `(cont $f)` and `$f` are different types. In particular, in the types section, you would separately have the following:
```
(type $f (func (param ...) (result ...)))
(type $c (cont $f))
```

Speaking just from an implementation perspective, having a dedicated `Continuation` type is mostly a matter of consistency and convenience: For example, in `HeapTypeInfo`, we may have a `HeapType continuation_func` field instead of `Continuation continuation` in the `union`. But that seems a bit dangerous: We must not accidentally mix up the `HeapType` that denotes the continuation itself vs. the function `HeapType` it is built from.

Further, having `Continuation` as its own type allowed me to add overloads for, say, the existing printing and hashing functions, and follow the same pattern as for other types when dispatching to these overloads.

Does that clarify things?

Reply:
Thanks! Now I understand.